ansible modules for IPMI, SOL, Server Power and LED locator

sandinak opened the pull request.

sandinak committed:
fixing libs to use /usr/bin/env python vs /usr/bin/python to support venv

sandinak committed:
fixing ansible code to be compatible with 1.x .. module declaration needs to be at top of file

Commit:
Merge pull request #1 from sandinak/compat_fixes

fixing ansible code to be compatible with 1.x .. module declaration n…

sandinak committed:
Merge remote-tracking branch 'upstream/master'

sandinak committed:
adding tools for sol and ipmi

sandinak committed:
Set the defult ssh port to 2400 .. but this will only be applied to CIMC 3.x

sandinak commented:
This has:
- cisco_imc_ipmi - enable and disable ipmi access
- cisco_imc_sol - enable and disable sol on the cimc
- cisco_imc_server - change power state and indicator_led

vvb commented on library/cisco_imc_ipmi.py:
```diff
@@ -0,0 +1,160 @@
+'''
+
+
+def login(module):
```
`login` and `logout` are implemented in common code. I see that you are using the below code later. so you can safely remove `login` and `logout` from this file.

```
from ansible.module_utils.cisco_imc import ImcConnection
conn = ImcConnection(module)                                 
server = conn.login()                                                 
conn.logout()
```

sandinak replied:
this was copied directly from cisco_imc_ntp. Probably should be cleaned up as well

vvb commented on library/cisco_imc_ipmi.py:
```diff
@@ -0,0 +1,160 @@
+    return server
+
+
+def logout(module, imc_server):
```
same as above

vvb commented on library/cisco_imc_ipmi.py:
```diff
@@ -0,0 +1,160 @@
+def setup_ipmi(server, module):
+    from imcsdk.apis.admin.ipmi import ipmi_disable
+    from imcsdk.apis.admin.ipmi import ipmi_enable
+    from imcsdk.apis.admin.ipmi import is_ipmi_enabled
```
@waghswapnil  this is more of a comment on API layer provided by imcsdk - `is_ipmi_enabled` -> `ipmi_is_enabled`

sandinak replied:
Yup .. if it's fixed we should update this as well.

vvb commented on library/cisco_imc_server.py:
```diff
@@ -0,0 +1,211 @@
+'''
+
+
+def login(module):
```
same as before

vvb commented on library/cisco_imc_server.py:
```diff
@@ -0,0 +1,211 @@
+    return server
+
+
+def logout(module, imc_server):
```
same as before

vvb commented on library/cisco_imc_server.py:
```diff
@@ -0,0 +1,211 @@
+        argument_spec=dict(
+            server_id=dict(required=False, type='int', default=1),
+            chassis_id=dict(required=False, type='int', default=1),
+            state=dict(required=False, type='str',
```
should this be `status`?

sandinak replied:
I don't' believe so.  status implies "position of affairs" .. state implies "particular condition at a time". Most of the ansible modules use 'state' as the moniker: file, service, apt, yum. The only one that uses 'status' AFAIK is the return value of 'stat'.

vvb replied:
well, the original comment was given because there were duplication of `state` variables.. now it makes sense.

vvb commented on library/cisco_imc_server.py:
```diff
@@ -0,0 +1,211 @@
+
+def setup_server(server, module):
+    ansible = module.params
+    status, locator_led = ansible["status"], ansible["locator_led"]
```
`locator_led` does not seem to exist in ansible parameters.

sandinak replied:
fixed

vvb commented on library/cisco_imc_server.py:
```diff
@@ -0,0 +1,211 @@
+    
+    if status is not None:
+        power_changed = setup_server_power(server, module, 
+                               status, timeout, interval)
```
where are `timeout` and `interval` coming from?

sandinak replied:
fixed and updated docs.

vvb commented on library/cisco_imc_server.py:
```diff
@@ -0,0 +1,211 @@
+        power_changed = setup_server_power(server, module, 
+                               status, timeout, interval)
+        
+    if locator_led is not None:
```
`check_mode` handling is not done here.

vvb commented on library/cisco_imc_sol.py:
```diff
@@ -0,0 +1,168 @@
+'''
+
+
+def login(module):
```
`login` method is not required

vvb commented on library/cisco_imc_sol.py:
```diff
@@ -0,0 +1,168 @@
+    return server
+
+
+def logout(module, imc_server):
```
`logout` method is not required

vvb reviewed (changes requested):
@sandinak thank you for writing up these modules. You have got most of them correct with a minor few nits. I have provided comments where due.

sandinak committed:
Updated 

- removed login and logout methods

sandinak committed:
Fixed per comments

- removed login and logout methods
- added timeout and interval to ansible 
- updated documentation 
- cleaned up check_mode
- fixed return value to include power state

sandinak committed:
Cleanup per comments

- removed login, logout methods
- added range check for ssh_port
- defaulted ssh_port to 2400 per CIMC defaults
- fixed check_mode 
- cleaned up change logic 
- return status as part of the result

sandinak committed:
Fixed default baud in the docs
